src/main/java/net/datafaker/providers/base/Barcode.java

```diff
@@ -67,23 +67,24 @@ private long ean(int length) {
             even = odd;
             odd = tmp;
         }
-        int var = 0;
-
-        if (length == 13) {
-            var = odd + even + (even << 1);
-        } else if (length == 8 || length == 14 || length == 12) {
-            var = odd + even + (odd << 1);
-        }
-
-        int rounded = roundToHighestMultiplyOfTen((var));
-        int checkDigit = rounded - var;
+        int output = calculateOutput(length, odd, even);
+        int rounded = roundToHighestMultiplyOfTen((output));
+        int checkDigit = rounded - output;
         int product = 10;
         while (product <= checkDigit) {
             product *= 10;
         }
         return firstPart * product + checkDigit;
     }
 
+    private int calculateOutput(int length, int odd, int even) {
+        return switch(length) {
+            case 13 -> odd + even + (even << 1);
+            case 8, 14, 12 -> odd + even + (odd << 1);
```

Comment:
```suggestion
            case 8, 12, 14 -> odd + even + (odd << 1);
```
if we are talking about improvement of readability then it's better to keep ordering

Reply:
Changed